index.js

```diff
@@ -1,18 +1,110 @@
-const readline = require('readline');
-const rl = readline.createInterface(process.stdin, process.stdout);
+/*-----readline module, promises API, interface for input/output ---------------*/
+const readline = require("readline");
+const readlineInterface = readline.createInterface(
+  process.stdin,
+  process.stdout
+);
 
 function ask(questionText) {
   return new Promise((resolve, reject) => {
-    rl.question(questionText, resolve);
+    readlineInterface.question(questionText, resolve);
   });
 }
-
+/*---------------------------------------*/
 start();
-
 async function start() {
-  console.log("Let's play a game where you (human) make up a number and I (computer) try to guess it.")
-  let secretNumber = await ask("What is your secret number?\nI won't peek, I promise...\n");
-  console.log('You entered: ' + secretNumber);
-  // Now try and complete the program.
+  let min = 1;
+  let max = null;
+  let ans = "N";
+  let guess = null;
+  let range = null;
+  count = 0;
+  minCache = [];
+  maxCache = [];
+
+  /*-------------choose game-----------------------------*/
+  game = await ask("Choose 1 to guess a number I'm thinking of; or 0 for me to guess a number you're thinking of");
+
+  /*------------------user guesses cpu's number--------------------*/
+  if (game === "1") {
+    max = await ask("upper limit? ");
+    console.log("OK! you picked " + max + " for the upper limit");
+
+    range = max - min + 1;
+    guess = Math.floor(Math.random() * range + min);
+    console.log(guess);
```

Comment:
You may want to get rid of this console.log() because it makes it really easy to win